examples/meanfield_lda.py

```diff
@@ -0,0 +1,175 @@
+    svi = SVI(model, guide, optimizer, loss=TraceEnum_ELBO(num_particles=1))
+    num_batches = int(math.ceil(docs.shape[0] / batch_size))
+
+    bar = trange(num_epochs)
+    for epoch in bar:
+        running_loss = 0.0
+        for i in range(num_batches):
+            idx = torch.arange(i * batch_size,
+                               min((i + 1) * batch_size, len(docs)))
+            batch_docs = docs[idx, :]
+            loss = svi.step(batch_docs.T, vocab_size,
+                            docs.shape[0], num_topics, idx)
+            running_loss += loss
+
+        epoch_loss = running_loss / docs.shape[0]
+        bar.set_postfix(epoch_loss='{:.2f}'.format(epoch_loss))
+
+
+def get_data(target_path, force_rewrite=False):
+    # Download David Blei's AP dataset
+    if not force_rewrite and not (Path(target_path) / 'ap.tgz').exists():
+        url = "http://www.cs.columbia.edu/~blei/lda-c/ap.tgz"
+        response = requests.get(url, stream=True)
+        if response.status_code == 200:
+            with (Path(target_path) / 'ap.tgz').open('wb') as f:
+                f.write(response.raw.read())
+
+        # Untar
+        tar = tarfile.open(Path(target_path) / 'ap.tgz', "r:gz")
+        tar.extractall(path=target_path)
+        tar.close()
+
+    # Load vocabulary in a dataframe
+    with (Path(target_path) / 'ap/vocab.txt').open('r') as f:
+        vocab = [x.strip() for x in f.readlines()]
+
+    vocab = pd.DataFrame(columns=['word'], data=vocab)
+    reserved = pd.DataFrame(columns=['word'],
+                            data=['blank, reserved to padding'])
+    vocab = reserved.append(vocab, ignore_index=True)
+    vocab['index'] = vocab.index
+
+    # Load documents
+    if not force_rewrite and (Path(target_path) / 'ap/docs.pt').exists():
+        docs = torch.load(Path(target_path) / 'ap/docs.pt')
+        return docs, vocab
+
+    with (Path(target_path) / 'ap/ap.txt').open('r') as f:
+        docs = [x.strip() for x in f.readlines() if not x.strip().startswith('<')]
+
+    # Tokenize docs
+    tokenizer = RegexpTokenizer(r'\w+')
+    for idx in range(len(docs)):
+        docs[idx] = docs[idx].lower()
+        docs[idx] = tokenizer.tokenize(docs[idx])
+
+    docs = [[token for token in doc if not token.isnumeric()] for doc in docs]
+    docs = [[token for token in doc if len(token) > 1] for doc in docs]
+
+    # Lemmatize docs
+    lemmatizer = WordNetLemmatizer()
+    docs = [[lemmatizer.lemmatize(token) for token in doc] for doc in docs]
+
+    # Substitute words by their indexes
+    for doc_id in range(len(docs)):
+        df = pd.DataFrame(columns=['word'], data=docs[doc_id])
+        df = pd.merge(df, vocab[['index', 'word']], how='left', on='word').dropna()
+        docs[doc_id] = torch.from_numpy(df['index'].astype(int).values)
+
+    # Remove docs with zero length (2 occurrences) and pad docs with 0 index
+    docs = [doc for doc in docs if len(doc) > 0]
+    docs = pad_sequence(docs, batch_first=True, padding_value=0)
```

Comment:
are you sure that padding each document with zeros is a good idea? doesn't that affect the performance of the model?

Reply:
@slayeren this implementation uses dense tensors and Pyro's automatic enumeration to integrate out `zeta` and is indeed quite slow (hence the title :) ).  See the [ProdLDA tutorial](http://pyro.ai/examples/prodlda.html) for a less naive approach to topic modeling in Pyro.

Please use our [forum](https://forum.pyro.ai/) for any followup questions.